libraries/helpers.rb

```diff
@@ -80,7 +80,7 @@ def parsed_cloud_credentials_api_key
 
       def parsed_target_hostname
         return new_resource.target_hostname if new_resource.target_hostname
-        node['cloud']['public_ipv4']
+        node['cloud']['public_ipv4'] rescue node['ipaddress']
```

Comment:
Could we use [chef-sugar's logic](https://github.com/sethvargo/chef-sugar/blob/master/lib/chef/sugar/ip.rb#L30-L41) for this? e.g.:
```
def parsed_target_hostname
  return new_resource.target_hostname if new_resource.target_hostname
  require 'chef/sugar'
  Chef::Sugar::IP.best_ip_for(node, node)
```

We already use this logic in a bunch of different places. If this cookbook doesn't depend on it, we can add the dependency.